File: Source/Fantasy.Base/Tracking/ITrackListener.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Fantasy.Tracking
{
    public interface ITrackListener : IDisposable
    {
        Guid Id { get; }
        string Name { get; }
        string Category { get; }

        object this[string name] { get; }

        string[] PropertyNames { get; }

        TrackFactory Connection { get; }

        bool IsActived { get;}

        event EventHandler ActiveStateChanged;

        event EventHandler<TrackChangedEventArgs> Changed;
    }
}

File: AutoConverted/Fantasy.Base/Tracking/ITrackListener.java
package Fantasy.Tracking;

public interface ITrackListener extends IDisposable
{
	Guid getId();
	String getName();
	String getCategory();

Object getItem(String name);

	String[] getPropertyNames();

	TrackFactory getConnection();

	boolean getIsActived();

//C# TO JAVA CONVERTER TODO TASK: Events are not available in Java:
//	event EventHandler ActiveStateChanged;

//C# TO JAVA CONVERTER TODO TASK: Events are not available in Java:
//	event EventHandler<TrackChangedEventArgs> Changed;
}